### PR TITLE
mailhog: 1.0.0 -> 1.0.1

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change
New upstream release https://github.com/mailhog/MailHog/releases/tag/v1.0.1

Also added a simple test.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
/nix/store/scbrzia1k90bmmlb8jpzhh6ckx1fcnq5-MailHog-1.0.0	  43.1M
/nix/store/n76p7lr52agz7zw2zksr73vhgyayjdci-MailHog-1.0.1	  43.3M
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/mail/mailhog/default.nix [hunk 1]
```diff
@@ -11,15 +11,13 @@ buildGoPackage rec {
...
     repo = "MailHog";
-    sha256 = "0r6zidkffb8q12zyxd063jy0ig2x93llna4zb5i2qjh9gb971i83";
+    sha256 = "124216850572r1h0ii7ad6jd1cd5czcvkz7k2jzvjb4pv2kl8p3y";
   };
 
-  goDeps = ./deps.nix;
-
   meta = with stdenv.lib; {
     description = "Web and API based SMTP testing";
     homepage = "https://github.com/mailhog/MailHog";
```
Comment (drewrisinger) [inline pkgs/servers/mail/mailhog/default.nix:19]:
```suggestion
    homepage = "https://github.com/mailhog/MailHog";
    changelog = "https://github.com/mailhog/MailHog/releases/tag/v${version}";
```

**Review by drewrisinger — CHANGES_REQUESTED** [review]
* [x] Diff LGTM
* [x] Commits LGTM
* [x] Builds via ``nix-review``
* [x] Missing changelog ``meta.changelog``
* [ ] Didn't/don't know how to run the test.

```
https://github.com/NixOS/nixpkgs/pull/97445
1 package built:
mailhog
```

File: pkgs/servers/mail/mailhog/default.nix (hunk 1 above)
Comment (jojosch) [reply pkgs/servers/mail/mailhog/default.nix:19]: added the changelog entry

**Comment by jojosch** [comment]
@drewrisinger the test can be run with `nix-build nixos/tests/mailhog.nix`

File: pkgs/servers/mail/mailhog/default.nix [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 buildGoPackage rec {
   pname = "MailHog";
-  version = "1.0.0";
+  version = "1.0.1";
   rev = "v${version}";
```
Comment (drewrisinger) [inline pkgs/servers/mail/mailhog/default.nix:6]:
```suggestion
```
This is non-standard. This line should be kept, moved into fetchFromGitHub.

**Review by drewrisinger — CHANGES_REQUESTED** [review]
Small change to Diff about fetcher rev, then LGTM.

I tried to run the test, but got the following error, not worth debugging IMHO. Would appreciate a merger to look over the test before merging, b/c I have no experience w/ nixos tests

```
error: a 'x86_64-linux' with features {kvm, nixos-test} is required to build '/nix/store/s52b23f87zr48j8h6srbiq5z09i17blk-vm-test-run-mailhog.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, nixos-test}
```

**Comment by drewrisinger** [comment]
Was just browsing through the meta section of manual, and that might suggest adding ``meta.passthru.tests``. See https://nixos.org/manual/nixpkgs/stable/#sec-standard-meta-attributes

I've honestly never dealt with nixos tests, so sorry I'm not much help.

**Review by doronbehar — CHANGES_REQUESTED** [review]
The commit should be separated into two - one adding a test and the other for the update.

**Comment by doronbehar** [comment]
Result of `nixpkgs-review pr 97445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mailhog</li>
  </ul>
</details>

**Comment by doronbehar** [comment]
Test runs fine as well.

**Pull request merged**